### ISSUE TITLE
`asdf plugin add` should exit code 0 in most cases

### ISSUE THREAD

**Issue by ghostsquad** [asdf-vm/asdf#841]
**Is your feature request related to a problem? Please describe.**
`asdf plugin add` exits with code 2 if the plugin is already added. This isn't a great user experience for automation. From the perspective of the final result, the operation succeeded.

**Describe the solution you'd like**
Exit with code `0` if the plugin is already installed. Possibly check/warn if the current plugin is outdated?

**Describe similar asdf features and why they are not sufficient**
n/a

**Describe workarounds you've considered**
```
asdf plugin add <name> || true
```

unfortunately this workaround is problematic in that if it fails for a different and legitimate reason, the script will not be aware.

**Additional context**
n/a

**Comment by jthegedus** [comment]
@ghostsquad We have had a prior discussion of this with #747

I **believe** a concensus was reached whereby we would remove non-zero exit codes on commands that did not fail, but did not change the state of things (due to an existing plugin as you suggest). As noted in https://github.com/asdf-vm/asdf/issues/747#issuecomment-691772544 the Issues and PRs that instituted this behaviour did not receive much feedback from our userbase (which was admittedly smaller) at the time.

@asdf-vm/core Are there any objections to addressing this issue with the proposition here https://github.com/asdf-vm/asdf/issues/747#issuecomment-706030258

**Comment by cccCody** [comment]
Totally agree. At least give us an option (e.g. similar to [`--failure-exit-code` in rspec](https://relishapp.com/rspec/rspec-core/docs/command-line/failure-exit-code-option-exit-status))

**Comment by ghostsquad** [comment]
I made a comment in the other issue, but it looks like it was closed. My code is littered with `asdf plugin add <blah> || true` and error handling suffers because of it, because if there's a problem getting the plugin, that error is essentially ignored.

**Comment by hyperupcall** [comment]
This is a popular request and have linked a PR that resolves this - hopefully it lands in `v0.12.1` soon.

For now, the best workaround is manually checking exit codes:

```sh
asdf_plugin_add() {
  asdf plugin add "$@" || {
    local exit_code=$1

    # If asdf detects that the plugin is already installed, it prints to standard error,
    # then exits with a code of 2. We manually check this case, and return 0 so this
    # this script remains idempotent, especially when `errexit` is set.
    if ((exit_code == 2)); then
      return 0
    else
      return $exit_code
    fi
  }
}

asdf_plugin_add <blah>
```
